File: chapter02/main/src/main/java/game/ComputerPlayer.java
package game;

import java.util.List;

public class ComputerPlayer extends Player {

  void fillCards(Deck deck) {
    while (this.getCardSum() <= 17) {
      this.addCard(deck.drawCard());
    }
  }

  @Override
  public List<Card> getCards() {
    if (isDrawing()) {
      return super.getCards().subList(0, 2);
    } else {
      return super.getCards();
    }
  }
}

File: chapter02/main/src/main/java/game/ComputerPlayer.kt
package game

import kotlin.math.min

class ComputerPlayer : Player() {
    internal fun fillCards(deck: Deck) {
        while (super.cardSum <= 17) {
            this.addCard(deck.drawCard())
        }
    }

    override fun showCards(): String {
        val showCards = if (isDrawing) {
            cards.subList(0, min(cards.size, 2))
        } else {
            cards
        }
        val cardsInfo = showCards.joinToString(separator = " ")
        return if (isDrawing) {
            cardsInfo
        } else {
            "$cardsInfo($cardSum)"
        }
    }
}
